### DISCUSSION TITLE
Show off, how easy it is to integrate with payment method such a Stripe?

### DISCUSSION THREAD

**Discussion by subhendukundu** [nhost/nhost#17] [Ideas]
Nextjs and Supabase did an amazing demo on this.
https://github.com/vercel/nextjs-subscription-payments

We can do something similar?

Also, another demo would be an actual use case of https://hasura.io/blog/building-stateful-apps-using-serverless-postgres-and-hasura/

I am working on adding Stripe to http://analytics.cool.bio
If I am able to pull it off, I will create a PR.

**Comment by alveshelio** [comment]
Hi @subhendukundu,

Johan made a video about integrating Stripe with Nhost, you should check here: https://www.youtube.com/watch?v=1V6KKbRaLhM, maybe this helps.

**Reply by subhendukundu** [reply]
Yes I was talking about this one on discord. However I am talking about more of showcase also where people get an quick template to get started. I think with the no-code and everything going on, people would be more into getting a product up and running rather focusing on hardcore engineering.

**Reply by alveshelio** [reply]
I totally get you, having ready to start templates will greatly increase the changes of people get started with Nhost. I've created a template with Nhost and React Query and a bung of other stuff and I was planning on maybe create another template with urQL, so people would be able to easily start a project without too much effort.

Good job helping the community :)

**Reply by subhendukundu** [reply]
Of course! Always have been a fan of Nhost. :)

**Reply by nunopato** [reply]
Love this @subhendukundu! thanks for creating this discussion. Johan and I did some brainstorming today about how to better structure and implement specific integrations. In the case of Stripe, we could set up a DB migration (e.g., for adding a `stripe_customer_id` column) + event trigger on user create + serverless function for adding that user as a Stripe customer - Nhost would take care of all that and one would simply have to "enable" the Stripe integration. How does this sound to you?

**Reply by subhendukundu** [reply]
That would be amazing. Also, I think we should think about more of a robust solution. Not just sticking to one payment solution. I personally love Stripe and using it almost everywhere. Yet other developers/clients might have a different preference. So we should make something easy to add any payment methods. For example webhook solutions. Most of them have hooks.
The idea is to help them ship and add anything they want. More robust and less head-breaking with anything. So that developers can say "Nhost" to any requirements they get. 🙈

**Comment by subhendukundu** [comment]
Can everyone please add up the tools you use and love to see integration examples or use cases with Nhost?

I will go first.
1. Stripe/Square - Run Subscription Payments with Stripe, invoices
2. Sendgrid/Sendinblue - Send emails based on events
3. Mailchimp - Adds new users from Nhost Authentication to a specified Mailchimp audience.
4. Bitly - Shortens URLs
5. Push notifications? Firebase?
6. http://vitejs.dev/ example
7. https://razzlejs.org example

**Reply by nunopato** [reply]
8. Netlify 
9. Vercel

**Reply by mrinalwahal** [reply]
10. Razorpay for payments integration. Works in more countries, especially in ones where Stripe doesn't. Has more features and is cheaper. Works very well in India, which is a large market.
11. Making an ngrok style local http tunnel to expose API container publicly over the internet, for easier testing for things like production webhooks, and stuff. This is a long shot, but if we are creating a list, then why not.
